### ISSUE TITLE
Bug correction: Nimfa early stopping

### ISSUE THREAD

**Issue by cohenjer** [benchopt/benchmark_nmf#5]
When running the nimfa solver(s), the run is stopped very early and has a weird behavior for the very first iterations.

Something my be going on in nimfa which stops the algorithm early on, despite the explicit stopping criterion of nimfa having been set to values that should ensure no stopping happens from nimfa.

**Comment by tomMoral** [comment]
Can you share a specific run log with  `BENCHOTP_DEBUG=1`?

**Comment by cohenjer** [comment]
Here is an excerp from the log with debug:
```bash
 |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Calling solver nimfa[loss=euclidean,strategy=MU] with stop val: 0
/home/cohen/.conda/envs/benchopt_benchmark_nmf/lib/python3.8/site-packages/benchopt/config.py:75: UserWarning: BenchOpt config file /home/cohen/.config/benchopt.ini is with mode 644.
As it stores sensitive information such as the github token,
it is advised to use mode 600 (user rw only).
  warnings.warn(
    |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Calling with stop val: 1                        
/home/cohen/.conda/envs/benchopt_benchmark_nmf/lib/python3.8/site-packages/benchopt/config.py:75: UserWarning: BenchOpt config file /home/cohen/.config/benchopt.ini is with mode 644.
As it stores sensitive information such as the github token,
it is advised to use mode 600 (user rw only).
  warnings.warn(
    |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Calling with stop val: 2                        
/home/cohen/.conda/envs/benchopt_benchmark_nmf/lib/python3.8/site-packages/benchopt/config.py:75: UserWarning: BenchOpt config file /home/cohen/.config/benchopt.ini is with mode 644.
As it stores sensitive information such as the github token,
it is advised to use mode 600 (user rw only).
  warnings.warn(
    |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Calling with stop val: 3                        
/home/cohen/.conda/envs/benchopt_benchmark_nmf/lib/python3.8/site-packages/benchopt/config.py:75: UserWarning: BenchOpt config file /home/cohen/.config/benchopt.ini is with mode 644.
As it stores sensitive information such as the github token,
it is advised to use mode 600 (user rw only).
  warnings.warn(
    |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Exit with delta = -3.64e+03.                    
    |--nimfa[loss=euclidean,strategy=MU]: [DEBUG] - Exit with delta_objective = 3.71e+02 and n_eval=3.0e+00.
```

**Comment by tomMoral** [comment]
This is due to the `stopping_criterion`, which stops because there is no progress after 3 iterations.
Try adding a class variable `stopping_criterion = SufficientProgressCriterion(patience=10)` in the solver, with extra import `from benchopt.stopping_criterion import SufficientProgressCriterion`?

**Comment by cohenjer** [comment]
Seems like something else is going on.
I added patience=20, here is the convergence plot (running only nimfa and simulated data)
![output plot](https://user-images.githubusercontent.com/22053798/201079747-9231104f-e1a5-42cd-ac80-cfba4ad08f6d.png)
See how the loss bounces back. This could be a plotly bug on top of nimfa stopping too early, but my guess is that both issues are related...

**Comment by cohenjer** [comment]
Fixed by #13, was caused by max_iter=0 actually running without iterations bound
